### PR TITLE
chore: enable `github/array-foreach` rule for `eslint`

### PR DESCRIPTION
**Pull request by faustbrian**
Enables the https://github.com/github/eslint-plugin-github/blob/main/docs/rules/array-foreach.md rule mentioned in https://github.com/ethereumjs/ethereumjs-monorepo/issues/1935

### CODE REVIEW

**Comment by holgerd77** [comment]
Rebased this via UI.

**Comment by holgerd77** [comment]
Will leave this open for someone else to review and merge (think this roughly looks good though), to better confirm that we have a broader consensus here, but I can somewhat follow along with the reasoning of the linting rule and could live with this rule.

File: packages/client/test/net/protocol/libp2psender.spec.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ tape('[Libp2pSender]', (t) => {
     const sender = new Libp2pSender(conn[0])
     const receiver = new Libp2pSender(conn[1])
     receiver.on('status', (status: any) => {
+        console.log({status})
```
Comment (acolytec3) [inline packages/client/test/net/protocol/libp2psender.spec.ts:13]: Do we need this console log for something?  If not, let's remove.

**Review by acolytec3 — APPROVED** [review]
Looks good, excepting one console log.

**Review by gabrocheleau — APPROVED** [review]
Looks good to me! nice work

**Pull request merged**